src/select/__tests__/select.test.tsx

```diff
@@ -450,6 +451,162 @@ describe.each([false, true])('expandToViewport=%s', expandToViewport => {
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+        wrapper.findTrigger().keydown(KeyCode.down);
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.findDisabledReason()!.getElement()).toHaveTextContent(
+          'disabled reason'
+        );
+      });
+
+      test('has no aria-describedby by default', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions,
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.getElement()).not.toHaveAttribute('aria-describedby');
+      });
+
+      test('has no aria-describedby without disabledReason', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.getElement()).not.toHaveAttribute('aria-describedby');
+      });
+
+      test('has aria-describedby with disabledReason', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+                disabledReason: 'disabled reason',
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.getElement()).toHaveAttribute('aria-describedby');
+      });
+
+      test('has hidden element (linked to aria-describedby) with disabledReason', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+                disabledReason: 'disabled reason',
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.find('span[hidden]')!.getElement()).toHaveTextContent(
+          'disabled reason'
+        );
+      });
+
+      test('can not select disabled with reason option', () => {
+        const onChange = jest.fn();
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+                disabledReason: 'disabled reason',
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+        wrapper.selectOptionByValue('1');
+        expect(onChange).not.toHaveBeenCalled();
```

Comment:
Should we also assert if the dropdown closes or not?

Reply:
No. If the dropdown is closed, the select option is not shown in the screen anymore

Reply:
Can we write an explicit assertion (or a separate test maybe): "after clicking on the disabled option, the dropdown stays open"?

Reply:
We can, but how it could help us? If disabled option closes the dropdown, it means that `onChange` handler triggers as well, so this tests fails.

Reply:
My question is not much connected to `onChange` function but testing clicking on a disabled item in general.

Reply:
added a new test